### ISSUE TITLE
Asynchronous transactions

### ISSUE THREAD

**Issue by NameOfTheDragon** [Tigra-Astronomy/TA.ReactiveCommunications#2]
Enhance the `DeviceTransaction` class with a new method `WaitForCompletionAsync`.

This will allow asynchronous execution of code after the transaction has completed.

> Originally created at 28/01/2017 00:06:14 by Tim Long as a enhancement with severity major.

**Comment by NameOfTheDragon** [comment]
This code will likely be useful when implementing this:

```
#!csharp
        static Task<bool> AsTask(this WaitHandle waitHandle, int timeoutMilliseconds, CancellationToken cancellation)
            {
            return Task.Run(() =>
                {
                var s = WaitHandle.WaitAny(new[] {waitHandle, cancellation.WaitHandle}, timeoutMilliseconds) == 0;
                cancellation.ThrowIfCancellationRequested();
                return s;
                }, cancellation);
            }


```

**Comment by NameOfTheDragon** [comment]
Implemented in 0.3.0
